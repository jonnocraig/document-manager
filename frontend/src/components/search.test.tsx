import * as React from 'react';
import { shallow } from 'enzyme';
import Search from './search';

function setup(nextProps = {}) {
  const props = {
    filter:'',
    handleFilter: jest.fn(),
    ...nextProps
  };
  const wrapper = shallow(<Search {...props} />);

  return {
    props,
    wrapper
  };
}

describe('<Search />', () => {
  
  it('should match the snapshot', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('should fire handleFilter if input value changed', () => {
    const { props, wrapper } = setup();
    const event = { preventDefault: jest.fn(), target: { value: 'abc' }};
    wrapper.find('#filterDocuments').simulate('change', event);
    expect(props.handleFilter.mock.calls.length).toBe(1);    
  });
});
