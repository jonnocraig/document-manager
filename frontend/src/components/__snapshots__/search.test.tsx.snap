// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <List
    filter=""
    handleFilter={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <FormGroup
        tag="div"
      >
        <InputGroup
          tag="div"
        >
          <InputGroupAddon
            addonType="prepend"
            tag="div"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="search"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Search"
              transform={null}
            />
          </InputGroupAddon>
          <Input
            id="filterDocuments"
            onChange={[MockFunction]}
            placeholder="Search by name or type"
            type="text"
            value=""
          />
        </InputGroup>
      </FormGroup>,
      "tag": "form",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <InputGroup
          tag="div"
        >
          <InputGroupAddon
            addonType="prepend"
            tag="div"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="search"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title="Search"
              transform={null}
            />
          </InputGroupAddon>
          <Input
            id="filterDocuments"
            onChange={[MockFunction]}
            placeholder="Search by name or type"
            type="text"
            value=""
          />
        </InputGroup>,
        "tag": "div",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <InputGroupAddon
              addonType="prepend"
              tag="div"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="search"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Search"
                transform={null}
              />
            </InputGroupAddon>,
            <Input
              id="filterDocuments"
              onChange={[MockFunction]}
              placeholder="Search by name or type"
              type="text"
              value=""
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "addonType": "prepend",
              "children": <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="search"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Search"
                transform={null}
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "border": false,
                "className": "",
                "fixedWidth": false,
                "flip": null,
                "icon": "search",
                "inverse": false,
                "listItem": false,
                "mask": null,
                "pull": null,
                "pulse": false,
                "rotation": null,
                "size": null,
                "spin": false,
                "symbol": false,
                "title": "Search",
                "transform": null,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "id": "filterDocuments",
              "onChange": [MockFunction],
              "placeholder": "Search by name or type",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <FormGroup
          tag="div"
        >
          <InputGroup
            tag="div"
          >
            <InputGroupAddon
              addonType="prepend"
              tag="div"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="search"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Search"
                transform={null}
              />
            </InputGroupAddon>
            <Input
              id="filterDocuments"
              onChange={[MockFunction]}
              placeholder="Search by name or type"
              type="text"
              value=""
            />
          </InputGroup>
        </FormGroup>,
        "tag": "form",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <InputGroup
            tag="div"
          >
            <InputGroupAddon
              addonType="prepend"
              tag="div"
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="search"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title="Search"
                transform={null}
              />
            </InputGroupAddon>
            <Input
              id="filterDocuments"
              onChange={[MockFunction]}
              placeholder="Search by name or type"
              type="text"
              value=""
            />
          </InputGroup>,
          "tag": "div",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <InputGroupAddon
                addonType="prepend"
                tag="div"
              >
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="search"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Search"
                  transform={null}
                />
              </InputGroupAddon>,
              <Input
                id="filterDocuments"
                onChange={[MockFunction]}
                placeholder="Search by name or type"
                type="text"
                value=""
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "addonType": "prepend",
                "children": <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon="search"
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  symbol={false}
                  title="Search"
                  transform={null}
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "border": false,
                  "className": "",
                  "fixedWidth": false,
                  "flip": null,
                  "icon": "search",
                  "inverse": false,
                  "listItem": false,
                  "mask": null,
                  "pull": null,
                  "pulse": false,
                  "rotation": null,
                  "size": null,
                  "spin": false,
                  "symbol": false,
                  "title": "Search",
                  "transform": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "id": "filterDocuments",
                "onChange": [MockFunction],
                "placeholder": "Search by name or type",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
